control-plane/api-gateway/common/resources.go

```diff
@@ -400,37 +413,45 @@ func (s *ResourceMap) TranslateAndMutateHTTPRoute(key types.NamespacedName, onUp
 
 	consulRoute, ok := s.consulHTTPRoutes[consulKey]
 	if ok {
-		// remove from the consulHTTPRoutes map since we don't want to
-		// GC it in the end
-		delete(s.consulHTTPRoutes, consulKey)
 		mutated := mutateFn(&consulRoute.route, *translated)
+		if len(mutated.Parents) != 0 {
+			// if we don't have any parents set, we keep this around to allow the route
+			// to be GC'd.
+			delete(s.consulHTTPRoutes, consulKey)
+			s.consulMutations = append(s.consulMutations, &ConsulUpdateOperation{
+				Entry:    &mutated,
+				OnUpdate: onUpdate,
+			})
+		}
```

Comment:
these parent length checks basically mean that if we have no parents we've bound to we're not going to create a route, which allows a bit of a simplification around the cleanup logic -- basically we haven't bound a route to anything, so GC it